da/kaspa/kaspa.go

```diff
@@ -0,0 +1,394 @@
+	c.pubsubServer = pubsubServer
+
+	c.batchRetryDelay = c.config.RetryDelay
+	c.batchRetryAttempts = *c.config.RetryAttempts
+
+	// Apply options
+	for _, apply := range options {
+		apply(c)
+	}
+
+	metrics.RollappConsecutiveFailedDASubmission.Set(0)
+	return nil
+}
+
+// Start starts DataAvailabilityLayerClient instance.
+func (c *DataAvailabilityLayerClient) Start() error {
+	c.logger.Info("Starting Kaspa Data Availability Layer Client.")
+	c.ctx, c.cancel = context.WithCancel(context.Background())
+	// other client has already been set
+	if c.client != nil {
+		c.logger.Info("Kaspa client already set.")
+		return nil
+	}
+
+	mnemonic := os.Getenv(c.config.MnemonicEnv)
+	if mnemonic == "" {
+		return fmt.Errorf("mnemonic environment variable %s is not set or empty", c.config.MnemonicEnv)
+	}
+
+	client, err := client.NewClient(c.ctx, &c.config, mnemonic)
+	if err != nil {
+		return fmt.Errorf("error while establishing connection to DA layer: %w", err)
+	}
+	c.client = client
+	return nil
+}
+
+// Stop stops DataAvailabilityLayerClient.
+func (c *DataAvailabilityLayerClient) Stop() error {
+	c.logger.Info("Stopping Kaspa Data Availability Layer Client.")
+	err := c.client.Stop()
+	if err != nil {
+		c.logger.Error("Stopping Kaspa client", "err", err)
+	}
+	return nil
+}
+
+// GetClientType returns client type.
+func (c *DataAvailabilityLayerClient) GetClientType() da.Client {
+	return da.Kaspa
+}
+
+// SubmitBatch submits batch to DataAvailabilityLayerClient instance.
+func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultSubmitBatch {
+	blob, err := batch.MarshalBinary()
+	if err != nil {
+		return da.ResultSubmitBatch{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	c.logger.Debug("Submitting to da batch with size", "size", len(blob))
+	return c.submitBatchLoop(blob)
+}
+
+// submitBatchLoop tries submitting the batch, keep trying indefinitely.
+func (c *DataAvailabilityLayerClient) submitBatchLoop(dataBlob []byte) da.ResultSubmitBatch {
+	backoff := c.config.Backoff.Backoff()
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			return da.ResultSubmitBatch{}
+		default:
+			var txHash []string
+			var blobHash string
+			err := retry.Do(
+				func() error {
+					var err error
+					// TODO: differentiate between unrecoverable and recoverable errors.(https://github.com/dymensionxyz/dymint/issues/1416)
+					txHash, blobHash, err = c.client.SubmitBlob(dataBlob)
+					if err != nil {
+						metrics.RollappConsecutiveFailedDASubmission.Inc()
+						c.logger.Error("broadcasting batch", "error", err)
+						return err
+					}
+					return nil
+				},
+				retry.Context(c.ctx),
+				retry.LastErrorOnly(true),
+				retry.Delay(c.batchRetryDelay),
+				retry.DelayType(retry.FixedDelay),
+				retry.Attempts(c.batchRetryAttempts),
+			)
+			if err != nil {
+				err = fmt.Errorf("broadcast data blob: %w", err)
+				c.logger.Error(err.Error())
+				continue
+			}
```

Comment:
can't this logic be simplified like the celestia client where we simply retry forever (which is basically what happens here, only here we retry with backoff, and than retry again):

for reference:

```
daMetaData, err := c.submit(data)
			if err != nil {
				c.logger.Error("Submit blob.", "error", err)
				metrics.RollappConsecutiveFailedDASubmission.Inc()
				backoff.Sleep()
				continue
			}
```